### DISCUSSION TITLE
Bad post url returns archive instead of 404 error

### DISCUSSION THREAD

**Discussion by selaromdotnet** [PiranhaCMS/piranha.core#2076] [Q&A]
I searched the issues history for this issue, and found this one: https://github.com/PiranhaCMS/piranha.core/issues/261 which shows closed as done. however on my instance (version 11.1.0) if I go to /blog/non-existant-post-slug I see the regular /blog archive page.

Did I miss a step or is something else wrong here? thanks!

**Comment by tidyui** [comment]
This is most likely a regression as the routing has been completed rewritten since that issue was closed. Opened a new issue for it #2078

**Comment by selaromdotnet** [comment]
ah thanks for that, I'll watch that issue for updates!

I am having a related issue that might be worth looking at as part of this one, but if it's just something unsupported or something I'm doing wrong, let me know that'd be helpful to understand.

Basically I am hoping to add pages underneath the archive page, while still supporting the posts serving from that parent archive page.

Like, my archive page is /blog, so /blog/post-slug would resolve to the archive page and show the post, as it already does.

Now I'd like to have a page like /blog/search where this is a custom SearchPage, routed to a custom SearchPage handler/route, that reads from the query string to do a rudimentary search of posts by title.

I have already created the SearchPage model, SearchPage handler, and routed the SearchPage model to the handler.

If I create the Search page at the root, like /test-search-page and navigate to that path, I see my custom page for it that I routed.

However, if I put that page under the /blog archive page, so that the path is now /blog/test-search-page, that path returns a 404. If I misspell it, like /blog-testing-search-page I see the standard archive page (the issue originally reported in this thread). So the /blog/test-search-page is doing something, but when it's under the /blog path, it breaks.

Is this an unsupported operation? thanks again!

**Reply by tidyui** [reply]
As the routing resolves the pages before trying to resolve pages this should be totally fine. Pages are resolved “backwards” trying to find the deepest page first, the moving up through the segments.

However, please note that permalinks are not remapped if you move pages, the full hierarchy is stored on each instance, just to make sure that the page you’re trying to find existed.

**Reply by selaromdotnet** [reply]
thanks for the insight, this is indeed what I assumed and expected. the first time I tried it I did put the search page in a different place, then moved it to the right place, but I updated the slug to be the expected path...

Is permalink the same as the slug, or is there a separate place to put that? I did end up just deleted the sub search page, and recreating it correctly, so the slug is still correct as /blog/search but that path returns a page not found, which ironically tells me it found something, because a bad slug like /blog/whatevernotreal doesn't return 404 but instead returns the blog page...

so i feel like i'm missing a step here, even though I do appear to be doing it correctly. i'm going to see if I can find some time to pull the source and trace through the resolution and see if I can find any insight, thanks again for the help. 

if you think of anything else I can try do let me know, many thanks again!

**Reply by selaromdotnet** [reply]
aha! i got it, i guess I didn't delete and recreate the page, because I looked at the actual records in the DB and i had the old route /search which I mistakenly defined the first time i created it. 

I manually changed this to the /searchpage route for my custom page handler, and now it works, yay!!

is there any way to change this without manually updating the db (or I assume, deleting and recreating the page, which probably would have worked and saved me a lot of time!)?

either way, i'm back in business, thanks a bunch, this platform really rocks!
